index.rst

```diff
@@ -1,61 +1,311 @@
+
+  # wait for background processes to finish
+  await asyncio.gather(select_sources_task, preproc_task_1, preproc_task_2)
+
+  # Process full array intra/extra focal images
+  await mtcs.rem.mtaos.cmd_runWEP.set_start(
+      intraId=data_id_1, extraId=data_id_2
+  )
+
+  # issue corrections to the components
+  await mtcs.rem.mtaos.cmd_issueCorrections.start()
+
+In the code above we used the already existing functionality of the `MTCS`_ and `ComCam`_ classes.
+Note that the process would be similar if we where to use ``LSSTCam`` instead of ``ComCam``, simply replacing the `ComCam`_ class by the equivalent when that becomes available.
+This means, SAL Scripts can be written for ``ComCam`` in a way that will be reusable for the main camera.
+
+.. _MTCS: https://ts-observatory-control.lsst.io/user-guide/maintel/mtcs-user-guide.html
+.. _ComCam: https://ts-observatory-control.lsst.io/user-guide/maintel/CCCamera-user-guide.html
+
+.. _user-provided-optical-aberration:
+
+User-provided optical aberration
+--------------------------------
+
+In some conditions is may be useful to allow users to specify optical aberrations to be introduced to the system, e.g., add a certain amount of comma to the system.
+The MTAOS can easily receive a list of Zernike coefficients, pass them through the Optical Feedback Control and compute the offsets and bending modes required.
+
+The command would simply receive an array of wavefront errors and apply them to the system properly registering the correction internally.
+
+.. code-block:: py
+
+  import numpy as np
+
+  # Assuming the wavefront correction is an array of 19 Zernike coefficients
+  wf = np.zeros(19)
+
+  # Assuming ANSI indexes, wf[7] is vertical comma
+  # Adding 1mm of vertical comma
+  wf[7] = 1.
+
+  await mtaos.cmd_runOFC.set_start(wf = wf)
+
+When the ``MTAOS`` is configured it will load appropriate configuration files for the OFC.
+During commissioning it is likely we will want to run the ``OFC`` with a different set of configurations.
+In order to allow quick turnaround, the command can accept an additional yaml string with overriding  parameters for the currently loaded configuration.
+
+For instance, one could truncate the sensitivity matrix so that a wavefront error correction is applied to a specific component (or components), e.g.;
+
+.. code-block:: py
+
+  import yaml
+
+  # switch off all corrections except m2 hexapod
+  # See https://github.com/lsst-ts/ts_ofc/blob/master/policy/zkAndDofIdxArraySet.yaml
+  ofc_config = {
+    "dofIdx": {
+        "m2Hex": [1, 1, 1, 1, 1],
+        "camHex": [0, 0, 0, 0, 0],
+        "m1m3Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        "m2Bend": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      }
+  }
+
+  # convert dictionary to yaml string
+  config = yaml.safe_dump(ofc_config)
+
+  await mtaos.cmd_runOFC.set_start(wf = wf, config = config)
```

Comment:
If you really want to do the overwritten by this method, you may need to use a clear configuration name instead of `config` here. There are many configuration parameters in `ts_wep` and `ts_ofc`. It will not be able for `ts_MTAOS` to judge which configuration data in `ts_wep` and `ts_ofc` to be rewritten. This is very difficult to maintain in `ts_MTAOS` level with this interface. We may need to have more nested architecture to let `ts_MTAOS` to know which parameter data to overwrite in `ts_wep` or `ts_ofc` level.